libs/api/libbpf_program.cpp

```diff
@@ -206,7 +206,7 @@ bpf_program__attach_xdp(const struct bpf_program* program, int ifindex)
     }
 
     bpf_link* link = nullptr;
-    ebpf_result_t result = ebpf_program_attach(program, &EBPF_ATTACH_TYPE_XDP, &ifindex, sizeof(ifindex), &link);
+    ebpf_result_t result = ebpf_program_attach(program, &EBPF_ATTACH_TYPE_XDP_TEST, &ifindex, sizeof(ifindex), &link);
```

Comment:
I think this change is wrong.  `bpf_program__attach_xdp()` is defined to use attach type XDP.
If you need something to use another attach type it would be a different function.  Same for any other `_xdp_ apis` in this file.

Since the current proposed change seems like it would break real XDP when used with xdp-for-windows, I consider this comment (unlike my other comments) a blocking comment, hence my "request changes".

Reply:
Please take a look at the latest iteration - Does this sufficiently address the blocking nature of the comment?

Reply:
I think I see other instances of this that should remain using the XDP guid. I will follow up with another iteration addressing this.